### PR TITLE
feat(frontend): introduce frontend for testing public API

### PR DESCRIPTION
**Pull request by Chreggii**

### CODE REVIEW

File: frontend/src/app/app.component.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<h2>File Upload</h2>
+<input type="file" (change)="onFileSelected($event)">
+
+<h2>Test Get Request</h2>
+<form [formGroup]="form">
+  <div>
+    <label>SourceCode:</label>
+    <input type="text" formControlName="sourceText">
```
Comment (olikami) [inline frontend/src/app/app.component.html:8]: This should be a `<textarea>`.

File: frontend/src/app/app.component.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  </div>
+  <div>
+    <label>Language:</label>
+    <input type="text" formControlName="language">
```
Comment (olikami) [inline frontend/src/app/app.component.html:12]: This should probably be a `<select>`

**Review by olikami — CHANGES_REQUESTED** [review]

File: frontend/src/app/app.component.html (hunk 1 above)
Comment (Chreggii) [reply frontend/src/app/app.component.html]: fixed

File: frontend/src/app/app.component.html (hunk 2 above)
Comment (Chreggii) [reply frontend/src/app/app.component.html:14]: Otherwise we can not test the error case

**Review by olikami — APPROVED** [review]

**Pull request merged**